counterpartylib/lib/messages/versions/enhanced_send.py

```diff
@@ -0,0 +1,202 @@
+      problems.append('memo is too long')
+
+    return problems
+
+def compose (db, source, destination, asset, quantity, memo, memo_is_hex):
+    cursor = db.cursor()
+
+    # Just send BTC?
+    if asset == config.BTC:
+        return (source, [(destination, quantity)], None)
+
+    # resolve subassets
+    asset = util.resolve_subasset_longname(db, asset)
+
+    #quantity must be in int satoshi (not float, string, etc)
+    if not isinstance(quantity, int):
+        raise exceptions.ComposeError('quantity must be an int (in satoshi)')
+
+    # Only for outgoing (incoming will overburn).
+    balances = list(cursor.execute('''SELECT * FROM balances WHERE (address = ? AND asset = ?)''', (source, asset)))
+    if not balances or balances[0]['quantity'] < quantity:
+        raise exceptions.ComposeError('insufficient funds')
+
+    # convert memo to memo_bytes based on memo_is_hex setting
+    if memo is None:
+        memo_bytes = b''
+    elif memo_is_hex:
+        memo_bytes = bytes.fromhex(memo)
+    else:
+        memo = memo.encode('utf-8')
+        memo_bytes = struct.pack(">{}s".format(len(memo)), memo)
+
+    block_index = util.CURRENT_BLOCK_INDEX
+
+    problems = validate(db, source, destination, asset, quantity, memo_bytes, block_index)
+    if problems: raise exceptions.ComposeError(problems)
+
+    asset_id = util.get_asset_id(db, asset, block_index)
+
+    short_address_bytes = address.pack(destination)
+
+    data = message_type.pack(ID)
+    data += struct.pack(FORMAT, asset_id, quantity, short_address_bytes)
+    data += memo_bytes
+
+    cursor.close()
+    # return an empty array as the second argument because we don't need to send BTC dust to the recipient
+    return (source, [], data)
+
+def parse (db, tx, message):
+    cursor = db.cursor()
+
+    # Unpack message.
+    try:
+        unpacked = unpack(db, message, tx['block_index'])
+        asset       = unpacked['asset']
+        quantity    = unpacked['quantity']
+        destination = unpacked['address']
+        memo_bytes  = unpacked['memo']
+
+        status = 'valid'
+
+    except (exceptions.UnpackError, exceptions.AssetNameError, struct.error) as e:
+        asset, quantity, destination, memo_bytes = None, None, None, None
+        status = 'invalid: could not unpack ({})'.format(e)
+    except:
+        asset, quantity, destination, memo_bytes = None, None, None, None
+        status = 'invalid: could not unpack'
+
+    if status == 'valid':
+        # Oversend
+        cursor.execute('''SELECT * FROM balances \
+                                     WHERE (address = ? AND asset = ?)''', (tx['source'], asset))
+        balances = cursor.fetchall()
+        if not balances:
+            status = 'invalid: insufficient funds'
+        elif balances[0]['quantity'] < quantity:
+            quantity = min(balances[0]['quantity'], quantity)
+
+    # For SQLite3
+    if quantity:
+        quantity = min(quantity, config.MAX_INT)
```

Comment:
tbh I wouldn't be opposed to taking the chance to breaking this "auto correcting" behaviour for enh sends ... (both for when quantity > MAX-INT and when quantity < balance), lets discuss on slack a bit maybe?